docs(guide): add Kubescape-cli guide. (#46)
kubescape/kubescape.io: docs/docs/use-cases/Kubescape-cli/kubescape-cli.md

### Install K3s

First, install K3s. Use the  --write-kubeconfig-mode 644 flag to allow Kubescape access to the cluster. 

```curl -sfL https://get.k3s.io | sh -s - --write-kubeconfig-mode 644```

Now, configure the KUBECONFIG environment variable to point to the default location of the K3s configuration:

```export KUBECONFIG=/etc/rancher/k3s/k3s.yaml```

Verify the cluster is up:

![image](01-cluster-up.png)

## Deploy a Pod
For this deployment, we will use a very simple NGINX container. 

### Create the Manifest

```touch nginx-app.yml```

#### nginx-app.yml

```
---
apiVersion: v1
kind: Namespace
metadata:
  name: app
  labels:
    name: app
---
apiVersion: v1
kind: Pod
metadata:
  name: nginx-app
  namespace: app
spec:
  containers:
  - name: nginx
    image: nginx:1.21
    ports:
    - containerPort: 80
    imagePullPolicy: Always
    securityContext:
        privileged: true
```

### Deploy the Pod
#### Deploy
```
kubectl apply -f nginx-app.yml
```
#### Verify
```
kubectl get pods -n app
```
![image](02-get-pods.png)

Great! Now that your cluster is set up and your pod is deployed, use Kubescape to harden it.

> ## Use Kubescape CLI

Reviewer: (another nit: "Kubescape CLI" isn't a brand name, it's "the Kubescape CLI" or "the Kubescape command-line tool")

> ## Use Kubescape CLI

Author: The subsections explained each piece of what we'd be doing, but I can add more. How is "Use the Kubescape CLI to Detect Control Failures"?

### Kubescape Scan
First, install Kubescape. Do so by running the script below:
```
curl -s https://raw.githubusercontent.com/kubescape/kubescape/master/install.sh | /bin/bash
```
Now, run the scan and view the output:
```
kubescape scan
```
![image](03-kubescape-scan-1.png)

Alright, the scan is finished. There’s a lot of output to go through. Using kubescape scan without any modifiers runs a scan using the MITRE and NSA frameworks. As you scroll down, you can see several of the controls scanned and their results. At the bottom, you’ll see your overall compliance score and the highest-stakes workloads:

![image](04-first-scan-summary.png)

> This guide will not focus on the Traefik control failures but will address the Pod/nginx-app issue soon. First, knock out one of the "Network" control failures. 

Author: This is something the users will see as it's meant to be followed. So even if we get a screenshot with it, the users will see it if they're following along.

> This guide will not focus on the Traefik control failures but will address the Pod/nginx-app issue soon. First, knock out one of the "Network" control failures. 

Reviewer: (Does K3s come with Traefik by default?)

> This guide will not focus on the Traefik control failures but will address the Pod/nginx-app issue soon. First, knock out one of the "Network" control failures. 

Author: Yep, it's the default configuration.

kind: Namespace
metadata:
  name: app
  labels:
    name: app
---
apiVersion: v1
kind: Pod
metadata:
  name: nginx-app
  namespace: app
spec:
  containers:
  - name: nginx-app
    image: nginx1.21
    ports:
    - containerPort: 80
    imagePullPolicy: Always
    securityContext:
        privileged: false
```
Since it's just a simple pod, delete and re-apply it:

```
kubectl delete -f nginx-app.yml && kubectl apply -f nginx-app.yml
```

Rerun the scan and you'll see that the `C0057` control is remediated as indicated by the green check:

```
kubescape scan workload Pod/nginx-app --namespace app
```

![image](09-privilege-remediation.png)

### Image Scanning
For this guide's final piece of the puzzle, you will scan the image and see if you can improve its security posture. At the bottom of the previous output, you should see "Receive a full report for localhost by running: `$ kubescape scan image nginx:1.21`". Run that command:
```
kubescape scan image nginx:1.21
```
![image](10-image-scan-1.png)

Alright, there are definitely some issues here. With some research, you’ll notice that 1.21 is not the latest version of NGINX. Upgrade that to the latest version:

```
nginx:latest
```

> After you have made the modification and saved the file, rebuild the container and re-apply the manifest:

Author: Another relic from the previous deployment. Sorry about that.

```
kubectl delete -f nginx-app.yml && kubectl apply -f nginx-app.yml
```

> Finally, run `kubescape scan nginx:latest` to see the results. The results shown in the image are current as of the writing of this post, so yours may differ:

Reviewer: ```suggestion
Finally, run `kubescape scan nginx:latest` to see the results. The results shown in the image are current as of the writing of this guide, so yours may differ:
```

![image](11-image-scan-2.png)

> Several vulnerabilities have been remediated! Unfortunately, not all of the vulnerabilities, but that's the nature of security! Your individual results may vary due to the number of open vulnerabilities for NGINX when you’re going through this guide. 

Reviewer: ```suggestion
Several vulnerabilities have been remediated! Unfortunately, not all of them, but that's the nature of security, and why you need a tool such as Kubescape. Your individual results will vary due to the current version and number of open vulnerabilities in NGINX when you’re going through this guide.
```

## Conclusion

> In this guide, you have used Kubescape to scan a basic Kubernetes cluster for vulnerabilities, researched those vulnerabilities, and remediated them with a very straightforward process thanks to the tools and documentation provided by ARMO. Kubescape can also scan Helm manifests, Kustomize directories, and more. While this guide only scratches the surface, you can leverage Kubescape CLI to harden your clusters to meet any security requirements you may have. 

Reviewer: ```suggestion
In this guide, you have used Kubescape to scan a basic Kubernetes cluster for vulnerabilities, researched those vulnerabilities, and remediated them with a very straightforward process. Kubescape can also scan Helm manifests, Kustomize directories, and more. While this guide only scratches the surface, you can leverage Kubescape's CLI to harden your clusters to meet your own security requirements.
```

## Deploy a Cluster
In this example, you will use a Kubernetes cluster deployed with K3s. Any cluster will do as long as it supports a local container image registry and basic Kubectl commands. If you don’t have a cluster meeting those criteria, you can set one up using the following instructions.

### Install K3s

First, install K3s. Use the  --write-kubeconfig-mode 644 flag to allow Kubescape access to the cluster. 

```curl -sfL https://get.k3s.io | sh -s - --write-kubeconfig-mode 644```

Now, configure the KUBECONFIG environment variable to point to the default location of the K3s configuration:

```export KUBECONFIG=/etc/rancher/k3s/k3s.yaml```

Verify the cluster is up:

![image](01-cluster-up.png)

Forward the port to 4000 since the Flask server will use 5000. Feel free to modify this if you wish. 

## Deploy a Pod

> For this deployment, we will use a very simple Flask container. 

Reviewer 2: I think it is not Flask anymore
